services/workflows-service/src/transaction/transaction.service.ts

```diff
@@ -43,19 +45,24 @@ export class TransactionService {
           correlationId: transaction.transactionCorrelationId,
         });
       } catch (error) {
-        let errorToLog: Error = new Error('Unknown error', { cause: error });
+        if (mappedTransactions.length === 1) {
+          throw error;
+        }
 
+        let errorMessage = 'Unknown error';
         if (
-          error instanceof Prisma.PrismaClientKnownRequestError &&
-          error.code === PRISMA_UNIQUE_CONSTRAINT_ERROR
+          (error as Prisma.PrismaClientKnownRequestError).name === 'PrismaClientKnownRequestError'
         ) {
-          errorToLog = new Error('Transaction already exists', { cause: error });
+          errorMessage = getErrorMessageFromPrismaError(
+            error as Prisma.PrismaClientKnownRequestError,
+          );
         } else {
```

Comment:
With this you won't need the assertions or depend on `name`.
```
export const isPrismaClientKnownRequestError = (error: unknown): error is Prisma.PrismaClientKnownRequestError => {
  return error instanceof Prisma.PrismaClientKnownRequestError;
}
```

Reply:
`instanceof` was used before and didnt work, I guess it related to the Prisma way initiates the class